### PR TITLE
update malt/build module

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

Fixes https://github.com/nf-core/createtaxdb/issues/29 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/malt/build/tests/main.nf.test [hunk 1]
```diff
@@ -59,13 +64,16 @@ nextflow_process {
     }
 
     test("sarscov2 - fastq - stub") {
-
+        config "./nextflow.config"
```
Comment (mahesh-panchal) [inline modules/nf-core/malt/build/tests/main.nf.test:67]:
```suggestion
```

**Comment by LilyAnderssonLee** [comment]
@jfy133 Amazing debugging methods and excellent solutions!!!

**Comment by jfy133** [comment]
Java _shakes fist_ - mostly thanks to @mahesh-panchal as always though :grimacing:

File: modules/nf-core/malt/run/tests/nextflow.config [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-
-    withName: MALT_RUN {
-        ext.args = "-m BlastN -J-Xmx8G -a . -f SAM"
-        memory = 12.GB
-    }
-
     withName: MALT_BUILD {
-        ext.args = "--sequenceType DNA"
+        ext.args = "-J-Xmx6G --sequenceType DNA"
+        memory   = 6.GB
```
[the remark this answers is not in the document]
Comment (LilyAnderssonLee) [reply modules/nf-core/malt/run/tests/nextflow.config:5]: It seems that [process.resourceLimits ](https://www.nextflow.io/docs/latest/reference/process.html#process-resourcelimits)doesn't handle java memory according to the docs.

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply modules/nf-core/malt/run/tests/nextflow.config:5]: No, but that should likely be set using a closure to access task.memory, and then the actual memory setting is limited by `process.resourceLimits`

[the remark this answers is not in the document]
Comment (LilyAnderssonLee) [reply modules/nf-core/malt/run/tests/nextflow.config]: I have updated. Could you double check if I did right?

File: modules/nf-core/malt/run/tests/nextflow.config [hunk 3]
```diff
@@ -1,13 +1,16 @@
 process {
+    resourceLimits = [
+        memory: '6.GB'
+    ]
```
Comment (mahesh-panchal) [inline modules/nf-core/malt/run/tests/nextflow.config:4]:
```suggestion
```
So this shouldn't be defined here, but in the profile that determines where these tests are run.

**Review by mahesh-panchal — COMMENTED** [review]
So I'm not sure if `process.resourceLimits` are defined for the runners, but they should be I think. Not here though

**Pull request merged**